tests/utils/test_dag_processing.py

```diff
@@ -138,6 +138,29 @@ def test_max_runs_when_no_files(self):
         child_pipe.close()
         parent_pipe.close()
 
+    def test_start_new_processes_with_files_to_exclude(self):
+        processor_factory = MagicMock()
+        manager = DagFileProcessorManager(
+            dag_directory='directory',
+            max_runs=1,
+            processor_factory=processor_factory,
+            processor_timeout=timedelta.max,
+            signal_conn=MagicMock(),
+            dag_ids=[],
+            pickle_dags=False,
+            async_mode=True)
+        f1 = '/tmp/f1.py'
+        f2 = '/tmp/f2.py'
+        manager._file_path_queue = [f1, f2]
+        files_paths_to_exclude_in_this_loop = {f1}
+
+        manager.start_new_processes(files_paths_to_exclude_in_this_loop)
```

Comment:
This is too tightly coupled to the way this is implemented, rather than the goal.

Also the fact that you _can't_ esaily test this probably means you need to re-think the a.pproach

Reply:
if the manager loop calls `prepare_file_path_queue()` in every loop, we don't need this change (passing `files_paths_to_exclude_in_this_loop` to the `start_new_processes`. 

It is only called when `if not self._file_path_queue`. If we remove this, we will also need to to some dupe work in the `prepare_file_path_queue`

Reply:
i chose to have `files_paths_to_exclude_in_this_loop` inside the `start_new_processes` as i was thinking that is single place we do a `pop(0)` and start new process, so adding a check will guarantee it won't double process dag files.